### PR TITLE
perf(github): replace three serial REST calls with single GraphQL query in get_prs_stats

### LINKED ISSUE #811: [NITPICK] github/stats.py:124-190 three serial gh api calls per get_prs_stats
## Finding

get_prs_stats makes three serial 'gh api search/issues' calls per invocation (total, merged, open). Could be parallel or use GraphQL with aliases; under-fetching the data with three round-trips is a perf nit, not a correctness one.

**Location:** `hephaestus/github/stats.py:124-190`

## Proposed fix

Replace with a single GraphQL query using aliases.

## Discovered via

`/hephaestus:repo-analyze-strict-full` 2026-05-29, S14

---
Tracking: #708

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary

Replaces three serial `gh api search/issues` REST calls in `get_prs_stats()` with a single GraphQL query using aliased fields. This eliminates two network round-trips while preserving all existing behavior and error handling.

**Changes:**
- Single GraphQL query with three aliased `search` fields (total, merged, open)
- jq filter coerces null fields to 0 before parsing
- Comprehensive error handling for malformed JSON and partial failures
- All return values computed from same base query

**Testing:**
- 7 new tests covering counts, failures, edge cases, and contract verification
- All 35 existing tests continue to pass
- Live smoke test against GitHub API succeeds

Closes #811

### CODE REVIEW

**Review by mvillmow — COMMENTED** [review]
POLICY VIOLATION: Closes (PR body empty — no Closes #811 line), auto-merge (enabled=true must be false during review), signed-commits (empty signing state array — no verified signatures).

File: hephaestus/github/stats.py [hunk 1]
```diff
@@ -139,48 +146,44 @@ def get_prs_stats(start_date: str, end_date: str, author: str | None, repo: str)
...
+            f"qTotal={base_query}",
+            "-f",
+            f"qMerged={base_query} is:merged",
+            "-f",
+            f"qOpen={base_query} state:open",
             "--jq",
-            ".total_count",
+            "[.data.total.issueCount // 0, "
+            ".data.merged.issueCount // 0, "
+            ".data.open.issueCount // 0]",
```
[the remark this answers is not in the document]
Comment (mvillmow) [reply hephaestus/github/stats.py]: Fixed: replaced the three adjacent implicit string literals (ISC001) with a single explicit jq filter string on one line. Commit 444ca7c.

**Review by mvillmow — COMMENTED** [review]
Fixed implicit string concatenation (ISC001): replaced the three adjacent jq filter string literals (lines 171-173) with a single explicit string on one line. All pre-commit hooks pass, 231 unit tests pass, commit is signed and verified (444ca7c).
